interface/ceed-qfunction.c

```diff
@@ -75,6 +76,85 @@ int CeedQFunctionCreateInterior(Ceed ceed, CeedInt vlength,
   return 0;
 }
 
+/**
+  @brief Default implementation of CeedQFunctionApplyIdentity.
```

Comment:
I set up this as a backend agnostic default.

Reply:
I guess I was concerned about a backend that would apply in blocks (without ever creating a full E-vector). You're currently managing this by creating short work vectors and still calling `CeedQFunctionApply`, but your non-QFunction C code can't be transferred to a device, for example.

If you like the current implementation, we can stick with it, but we'll want to keep this alternate in mind.

Reply:
I'm not sure I understand your concern. With this we use the backend COPY_VALUES, so it will be using device operations on the CUDA backend, for instance.

Reply:
If you take a look at the `cuda-gen` backend for instance, you never have access from the host to any E-vector nor Q-vector for instance.

Reply:
Ah, that makes sense. Though, I'd imagine that cuda-gen will want to do something smarter than run a meaningless qfunction, right?

Reply:
Yes, that's true, but I think @jedbrown has a point, especially considering that some backends are lagging on the development. Each of this new little feature also make writing a new backend from scratch more challenging. But I think it's okay if you don't want to have this default implementation, even though that's a small weakness from a design point of view.

Reply:
Well, run-time compilation with access to the identity code (as it does), may be able to elide it there. My concern is that we want backends to be able to fuse and schedule composite backends (e.g., mixed topology/mixed order meshes using CUDA Streams or Graphs), but this approach obliges the backend to either handle identity as a special case or to have the reference identity handling prevent that fusion.

Reply:
Yeah, I don't like my approach now (the fact that OCCA changed something since I last updated it locally and my approach doesn't work on OCCA doesn't help). I was trying to avoid dragging around a collection of the different identity qfunction formats. Really, it would be nice to do this after CUDA and CPU share a source and we start the qfunction gallery.

Reply:
I was able to fix the OCCA issue. I'm happy with everything but the generic implementation in here.